libtiledbsoma/src/soma/soma_dataframe.cc

```diff
@@ -249,7 +248,14 @@ void SOMADataFrame::update_dataframe_schema(
 
         se.add_attribute(attr);
     }
-    se.array_evolve(uri);
+
+    se.array_evolve(uri_);
+
+    // When we evolve the schema, the ArraySchema needs to be updated to the
+    // latest version so re-open the Array
+    auto mode = arr_->query_type();
+    arr_->close();
+    arr_->open(mode);
```

Comment:
How does this work if the array was opened at a specific timestamp previously?

Reply:
TileDB `Array::open` will open it at set timestamp.

Reply:
Please add a comment to that effect :)